Fixed error in model.forecast method. Fixes #8

danielkentwood opened the pull request:
changed line 135 in models:
`deque(X[(-1 - d):(-1 - q):-1, i])`
to the following:
`deque(X[(-1 - d):(-1 - d - q):-1, i])`

Also updated the import of sklearn metrics; sklearn.metrics.regression is deprecated.

danielkentwood committed:
Fixed error in model.forecast method

changed line 135 in models:
`deque(X[(-1 - d):(-1 - q):-1, i])`
to the following:
`deque(X[(-1 - d):(-1 - d - q):-1, i])`

Also updated the import of sklearn metrics; sklearn.metrics.regression is deprecated.

jxx123 commented:
Thank you so much for the contribution! The change looks good to me. Have you run the tests (especially fireTS/tests/test_forecast.py) just for a sanity check? Let me know if you are interested in adding a test point in fireTS/tests/test_forecast.py to test this specific change (by specifying non-zero delay).

danielkentwood committed:
Update test_forecast.py

added a test that ensures deterministic output of forecast method with varying exogenous delays

danielkentwood commented:
> Thank you so much for the contribution! The change looks good to me. Have you run the tests (especially fireTS/tests/test_forecast.py) just for a sanity check? Let me know if you are interested in adding a test point in fireTS/tests/test_forecast.py to test this specific change (by specifying non-zero delay).

Happy to contribute! All the current tests did pass. I also took a crack at writing a test that checks for deterministic output of the forecast method at several exogenous delay values.

jxx123 commented:
Thank you so much! I will merge the PR. I need to set up the CI in the future to make this process more smooth.
